plugin/pkg/scheduler/algorithm/priorities/reduce.go

```diff
@@ -0,0 +1,46 @@
+package priorities
+
+import (
+	"k8s.io/api/core/v1"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/algorithm"
+	schedulerapi "k8s.io/kubernetes/plugin/pkg/scheduler/api"
+	"k8s.io/kubernetes/plugin/pkg/scheduler/schedulercache"
+)
+
+// NormalizeReduce generates a PriorityReduceFunction that can normalize the result
+// scores to [0, maxPriority]. If reverse is set to true, it reverses the scores by
+// subtracting it from maxPriority.
+func NormalizeReduce(maxPriority int, reverse bool) algorithm.PriorityReduceFunction {
+	return func(
+		_ *v1.Pod,
+		_ interface{},
+		_ map[string]*schedulercache.NodeInfo,
+		result schedulerapi.HostPriorityList) error {
+
+		var maxCount int
+		for i := range result {
+			if result[i].Score > maxCount {
+				maxCount = result[i].Score
+			}
+		}
+
+		if maxCount == 0 {
+			if reverse {
+				for i := range result {
+					result[i].Score = maxPriority
+				}
+			}
+			return nil
+		}
+
+		for i := range result {
+			score := &result[i].Score
```

Comment:
It's not compulsory but I would suggest to avoid using reference here. The `* *score` below just looks a bit wired (maybe common in C++?).  And you can write back the value as we did in the original code before.

Reply:
not a C++ programmer.. just wanna save one line of code. will change it.